### PR TITLE
[ShapeOptimizationApplication] adapt to new vtk output

### PR DESCRIPTION
**Pull request by armingeiser**
To be merged **after** #3656! 

Uses the new vtk output from the core

### CODE REVIEW

File: applications/ShapeOptimizationApplication/python_scripts/design_logger_vtk.py [hunk 1]
```diff
@@ -6,71 +6,96 @@
...
+        return vtk_output_process.Factory(self.output_settings["output_format"], self.model)
 
     # --------------------------------------------------------------------------
     def InitializeLogging( self ):
-        self.VtkIO.InitializeLogging()
+        self.vtk_io.ExecuteInitialize()
+        self.vtk_io.ExecuteBeforeSolutionLoop()
 
     # --------------------------------------------------------------------------
     def LogCurrentDesign( self, optimizationIteration ):
```
Comment (dbaumgaertner) [inline applications/ShapeOptimizationApplication/python_scripts/design_logger_vtk.py:81]: I think its a good idea to stick to the style guide regarding the naming of variabels. E.g. optimizationIteration -> optimization_iteration (function argument) or below OriginalTime -> original_time (local variable).

File: applications/ShapeOptimizationApplication/test_examples/.gitignore [hunk 2]
```diff
@@ -1 +1 @@
-**/*.csv
\ No newline at end of file
+**/Optimization_Results/
```
Comment (dbaumgaertner) [inline applications/ShapeOptimizationApplication/test_examples/.gitignore]: Why is this necessary?

**Review by dbaumgaertner — CHANGES_REQUESTED** [review]
Thanks for adjusting this. Looks good to me. Only some minor naming things. 

Thanks to @adityaghantasala, we can finally drop our custom vtk io.

File: applications/ShapeOptimizationApplication/test_examples/.gitignore (hunk 2 above)
Comment (armingeiser) [reply applications/ShapeOptimizationApplication/test_examples/.gitignore]: The result name of the vtk files is not anymore "design_history.vtk" but "MODELPARTNAME.vtk". Tto avoid accidental commits of the result files, i think its the easiest solution to just exclude the complete optimization results directory from git.

Comment (dbaumgaertner) [reply applications/ShapeOptimizationApplication/test_examples/.gitignore]: Ah ok. But note that the name of the results directory is an option in the optimization settings under "output_directory". That would make this gitignore obsolete. But I think it even makes sense to remove that option and fix the name.

Another thing is the filename of the design results. Note, that this is also an option. However, this PR changes that logic for the VTK-output so that the name of the corresponding sub-mdpa is used. But also here, I think, we can remove that option and use the name of the sub-mdpa in all cases.

Last not least we might fix the "response_log_filename" so that this also drops out from the settings. By that, the output options would really reduce to the actually necessary things like nodal_results.

@armingeiser What do you think?

Comment (armingeiser) [reply applications/ShapeOptimizationApplication/test_examples/.gitignore]: :+1: completely agree. I think we never used these options since we have them.

**Comment by dbaumgaertner** [comment]
@armingeiser Ready for another review. Note that I took the chance to adjust all the primal settings to the new process format such that we do not get these warnings anymore. Please take a careful look on the changes I made with the tests. They should be only of cosmetic nature.

File: applications/ShapeOptimizationApplication/python_scripts/design_logger_unv.py [hunk 3]
```diff
@@ -23,42 +23,43 @@
...
-        if self.WriteDesignSurface:
-            self.UNVIO = UniversalFileIO( self.DesignSurface, DesignHistoryFilenameWithPath, "WriteConditionsOnly", NodalResults )
-        elif self.WriteOptimizationModelPart:
-            self.UNVIO = UniversalFileIO( self.OptimizationModelPart, DesignHistoryFilenameWithPath, "WriteElementsOnly", NodalResults )
+        results_directory = self.output_settings["output_directory"].GetString()
+        design_history_filename_with_path =  results_directory+"/"+self.design_history_filename
+        nodal_results = self.output_settings["nodal_results"]
+
+        if self.write_design_surface:
+            self.UNVIO = UniversalFileIO( self.design_surface, design_history_filename_with_path, "WriteConditionsOnly", nodal_results )
```
Comment (armingeiser) [inline applications/ShapeOptimizationApplication/python_scripts/design_logger_unv.py]: `design_history_filename_with_path` could be `design_history_filepath`

File: applications/ShapeOptimizationApplication/tests/algorithm_steepest_descent_test/run_test.py [hunk 4]
```diff
@@ -25,14 +25,14 @@
 # Test results and clean directory
 # =======================================================================================================
 output_directory = parameters["optimization_settings"]["output"]["output_directory"].GetString()
-response_log_filename = parameters["optimization_settings"]["output"]["response_log_filename"].GetString() + ".csv"
+optimization_log_filename = parameters["optimization_settings"]["output"]["optimization_log_filename"].GetString() + ".csv"
```
Comment (armingeiser) [inline applications/ShapeOptimizationApplication/tests/algorithm_steepest_descent_test/run_test.py:28]: Are the default settings added to this "parameters" variable? Or how can you get it from the Parameters object if they are not defined in the json file?

**Review by armingeiser — COMMENTED** [review]
Wow, thanks for updating all those json files. I have just minor comments.

File: applications/ShapeOptimizationApplication/tests/algorithm_steepest_descent_test/run_test.py (hunk 4 above)
Comment (dbaumgaertner) [reply applications/ShapeOptimizationApplication/tests/algorithm_steepest_descent_test/run_test.py:28]: The corresponding default settings are added the moment we construct the optimizer. But it is not very obvious here, I agree. For the tests, though, it should be fine.

**Review by dbaumgaertner — APPROVED** [review]
